build/commands/lib/util.js

```diff
@@ -606,48 +607,98 @@ const util = {
     }
 
     Log.progressStart('build redirect_cc')
-    gnArgs = {
+    const buildArgs = {
       'import("//brave/tools/redirect_cc/args.gni")': null,
       use_remoteexec: config.useRemoteExec,
       rbe_exec_root: config.rbeExecRoot,
       rbe_bin_dir: config.realRewrapperDir,
       real_rewrapper: path.join(config.realRewrapperDir, 'rewrapper'),
     }
 
-    const buildArgsStr = util.buildArgsToString(gnArgs)
-    util.run('gn', ['gen', config.nativeRedirectCCDir, '--args="' + buildArgsStr + '"'], options)
-
+    util.runGnGen(config.nativeRedirectCCDir, buildArgs)
     util.buildTarget('brave/tools/redirect_cc', mergeWithDefault({outputDir: config.nativeRedirectCCDir}))
     Log.progressFinish('build redirect_cc')
   },
 
-  runGnGen: (options) => {
-    const buildArgsStr = util.buildArgsToString(config.buildArgs())
-    const buildArgsFile = path.join(config.outputDir, 'brave_build_args.txt')
-    const buildNinjaFile = path.join(config.outputDir, 'build.ninja')
-    const gnArgsFile = path.join(config.outputDir, 'args.gn')
-    const prevBuildArgs = fs.existsSync(buildArgsFile) ?
-      fs.readFileSync(buildArgsFile) : undefined
-    const extraGnGenOptsFile = path.join(config.outputDir, 'brave_extra_gn_gen_opts.txt')
-    const prevExtraGnGenOpts = fs.existsSync(extraGnGenOptsFile) ?
-      fs.readFileSync(extraGnGenOptsFile) : undefined
+  runGnGen: (outputDir, buildArgs, extraGnGenOpts = [], options = config.defaultOptions) => {
```

Comment:
IMHO `extraGnGenOpts` is a little bit weird thing.
If it's for one time `gn gen` launch (with `--ide=vs`) it's enough to pass it something like that `--force_gn_gen=extraGnGenOpts`. Without any storing logic.

If we use pass `extraGnGenOpts` each time gn gen `is launched` we should cover the case when `gn gen` is called by `ninja`. No sure we really want it.

P.S. It isn't a new issue and out of scope the current PR.

Reply:
Hm, probably gn is able to store the extra args and reuse it:
`If you manually run "gen" again you will need to resupply this argument, but normally GN will keep the build and IDE files up to date automatically when you build.`
So it actually works well.

Reply:
It works only when `gn gen` is called as a ninja regeneration step. We can call it directly, which means we should always pass these extra options.